Archives/ClmFile.cpp

```diff
@@ -157,7 +157,19 @@ namespace Archives
 			filesToPack[i] = GetInternalFileName(i) + ".wav";
 		}
 
-		return CreateArchive("temp.clm", filesToPack);
+		std::string tempFileName = "temp.clm";
+		if (!CreateArchive(tempFileName, filesToPack)) {
+			return false;
+		}
+
+		// Rename the output file to the desired file
+		try {
+			ReplaceFileWithFile(m_ArchiveFileName, tempFileName);
+			return true;
+		}
+		catch (std::exception& e) {
+			return false;
+		}
```

Comment:
Huh, I'm surprised by this change. That seems like quite the glaring inconsistency between the Clm code and the Vol code. It looks like a big bug on my part, and I'm wondering now if I ever tested the code.

Well, a valiant effort then to correct shoddy code, complicated by the messy return codes.